docs/self-hosted-appcircle/install-server/kubernetes.md

```diff
@@ -0,0 +1,535 @@
+
+Ensure the **one certificate** covers **all the subdomains** in the [domain name](#domain-name) section.
+
+Currently, the self-hosted Appcircle does not support the use of password-protected private keys for SSL certificates. The SSL certificate should be in PEM format.
+
+Additionally, configure the Appcircle server with a **fullchain certificate**, which should include the leaf (or app) certificate, intermediate certificates, and the root certificate, establishing a complete and trusted certificate chain.
+
+:::tip
+You can use a **wildcard certificate** to cover all the subdomains, simplifying the certificate management process. For example, a wildcard certificate for **`*.appcircle.spacetech.com`** will be enough.
+:::
+
+:::caution
+If you use a domain like `appcircle.spacetech.com`, it will have **two levels of subdomains**. Ensure that both your DNS provider and SSL certificate provider support multi-level subdomains for proper configuration.
+:::
+
+:::tip
+Appcircle **supports TLS 1.3**, the latest and most secure version of the TLS protocol, ensuring improved performance and stronger encryption for your connections.
+:::
+
+### Kubernetes cluster
+
+To install the Appcircle server using Helm, a Kubernetes cluster is required. The cluster should meet the following hardware specifications:
+
+**Minimum hardware requirements for enterprise installation:**
+
+- Nodes with `x86_64` architecture
+- 8 CPUs
+- 16 GB RAM
+- 500 GB Disk
+
+**Recommended hardware requirements for enterprise installation:**
+
+- Nodes with `x86_64` architecture
+- 32 CPUs
+- 64 GB RAM
+- 1 TB Disk
+
+For production environments, if you deploy stateful applications with the Appcircle Helm chart, you will need significant storage capacity, as specified above. You can configure disk resource allocations through Helm values according to your needs.
+
+However, if you opt to use external services for components such as PostgreSQL or MinIO, the storage requirements for the cluster are significantly reduced to around 50GB.
+
+:::info
+Using SSD storage is highly recommended if stateful applications are installed within the Appcircle Helm chart scope. SSDs provide faster read/write speeds, improving the performance and responsiveness of your applications.
+:::
+
+:::tip
+For storage class details, you can check the [Storage Class Configuration](/self-hosted-appcircle/configure-server/kubernetes-configuration.md#persistent-volume-configuration) section.
+:::
+
+Additionally, ensure that your Kubernetes version is 1.29.1 or later to maintain compatibility and support.
+
+### Install `kubectl`
+
+The `kubectl` CLI configured for the target Kubernetes cluster is required.
+
+### Install Helm
+
+Helm version `3.11.0` or later is required for deployment.
+
+### Kubernetes Ingress Controller
```

Comment:
The case mentioned in the issue below will be added as an FAQ or caution box to the Ingress documents for Express or default installation method.
- https://linear.app/appcircle/issue/BE-4994/minio-signed-urls-should-be-created-for-external-domain